### ISSUE TITLE
 Learning pathways configuration issue

### ISSUE THREAD

**Issue by Darylball41** [pnp/custom-learning-office-365#524]
We are getting the "Microsoft learning pathways has a configuration issue. Ask your administrator for assistance." error. This error is intermittent, however when it does occur it appears to affect everybody trying to use LP.

It's difficult to say how to reproduce it, sometimes none of the Microsoft content/the admin centre will load and instead the error page will be displayed this happens on every install (We've tried deleting/re-adding LP). We provisioned the site using the lookbook each time.

I'm unable to share full logs, however the errors as shown in console are:
business.bing.com/work/api/v3/user/init:1 Failed to load resource: net::ERR_FAILED

uksouth1-sphomep.svc.ms/api/v1/documents/metadata:1 Failed to load resource: the server responded with a status of 500 ()

custom-learning-web-part_9f36b4992c3974449682ffe4b8867764.js:1 Message: .... M365LP:InitService (getManifest) - TypeError: Failed to fetch

pnp.github.io/custom-learning-office-365/learningpathways/v4//metadata.json:1 Failed to load resource: net::ERR_FAILED

custom-learning-web-part_9f36b4992c3974449682ffe4b8867764.js:1 Message: .... M365LP:DataService (getMetadata) - TypeError: Failed to fetch

custom-learning-web-part_9f36b4992c3974449682ffe4b8867764.js:1 Message: .... M365LP:CustomLearningWebPart (firstInit) - TypeError: Cannot read property 'TelemetryKey' of null -- Could not initialize web part.

activitymonitor.js:211 DOMException: Failed to execute 'atob' on 'Window': The string to be decoded is not correctly encoded.

activitymonitor.js:228 Error: Error: Idle tracking cookie is missing.

DevTools failed to load SourceMap: Could not load content for sharepoint site URL/apps/ClientSideAssets/ID String/en-us.js.map: HTTP error: status code 404, net::ERR_HTTP_RESPONSE_CODE_FAILURE

Note: URL removed above.

DevTools failed to load SourceMap: Could not load content for https://statica.akamai.odsp.cdn.office.net/bld/_layouts/15/16.0.21312.12002/ActivityMonitor.js.map: HTTP error: status code 404, net::ERR_HTTP_RESPONSE_CODE_FAILURE

DevTools failed to load SourceMap: Could not load content for https://amcdn.msftauth.net/meInlineBackCompat.min.js.map: HTTP error: status code 404, net::ERR_HTTP_RESPONSE_CODE_FAILURE

Learning Pathways version number: 4.2.0 (The same also occurs on 4.1.4)

**Comment by jellis-thrive** [comment]
Hi @Darylball41 

The logs suggest that security may be preventing connectivity to GitHub, for some if not all users. Can we verify this is not occurring?

**Comment by hhmaity** [comment]
We're having the same issue here . we are allowing some specific users to access the LP site via SharePoint Online Group and giving them reader access . Some are reporting the same issue above.

My permissions to the site are granted via a Microsoft 365 Group and im able to access the pages with no issues.

**Comment by Scu-OSN** [comment]
We have the same set-up and we were experiencing problems, too. In our case, the following thing helped:
give the user group read permissions on the Learning Pathways site **AND** the App Catalog site. 
Hope this will work for you, too!

**Comment by Darylball41** [comment]
> Hi @Darylball41
> 
> The logs suggest that security may be preventing connectivity to GitHub, for some if not all users. Can we verify this is not occurring?

@jellis-thrive The specific GitHub for LP has been whitelisted on our proxy, so the connection shouldn't be getting blocked.

@Scu-OSN SPO Unfortunately all internal users already have read access to the app catalog - But thanks for the advice, it was worth a check.

**Comment by jellis-thrive** [comment]
> @jellis-thrive The specific GitHub for LP has been whitelisted on our proxy, so the connection shouldn't be getting blocked.

This may not fit the security requirement for LP to pull content. I will work to confirm this.

**Comment by jellis-thrive** [comment]
Hi @Darylball41 

Could you detail what specifically was whitelisted? A URL, or URL(s), would be great.

**Comment by SusMiller** [comment]
Hi @jellis-thrive, @Darylball41 is on leave this week so I'll pick up on responding to this. 

The following 2 URLs were whitelisted:
 https://business.bing.com/work/api/v3/user/init
https://pnp.github.io/custom-learning-office-365/learningpathways

Are there others we need to look at? These were the only 2 the console showed as failing to load because of being blocked by CORS. Despite these being allowed through, we are still getting the same CORS errors intermittently.

**Comment by jellis-thrive** [comment]
Hi @SusMiller 

https://pnp.github.io/custom-learning-office-365/learningpathways is the primary location that needs to be whitelisted, yes. If you are still getting CORS errors with that URL on the whitelist, network traces and a discussion with your security folks would be a best next step. The current expectation to the best of my knowledge is a working LP install with that URL on the whitelist. I will confirm it is the requirement.

**Comment by thomasdaggett** [comment]
Any update on this issue, I am seeing this problem in our environment.  I also tried updating to the latest LP install but same problem persists.

**Comment by juliemturner** [comment]
@thomasdaggett -- seeing this error is unique to every environment.  Please open your own issue and try and fill out the issue as completely as possible to assist us in helping you. Note the debugging steps linked in the issue item regarding this error.

**Comment by SusMiller** [comment]
Thanks @jellis-thrive,
Just to update we're continuing to investigate our side with proxy/VPN but struggling to pin down the cause.

**Comment by dcashpeterson** [comment]
Closed due to inactivity. If this is still an issue please create a new ticket and reference this one.
